### ISSUE TITLE
[Linux/arm64] GitHub_20657 fails with JitStress=2/JitStressRegs=1

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#22253]
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm64_cross_checked_ubuntu16.04_jitstress2_jitstressregs1_tst/108/consoleFull

Tests:
```
./JIT/jit64/hfa/main/testC/hfa_nd1C_r/hfa_nd1C_r.exe
./JIT/jit64/hfa/main/testC/hfa_nd0C_r/hfa_nd0C_r.exe
./JIT/jit64/hfa/main/testC/hfa_sd0C_r/hfa_sd0C_r.exe
./JIT/jit64/hfa/main/testC/hfa_sd1C_r/hfa_sd1C_r.exe
```

Tests appear to pass, but return code 1, not 100.

**Comment by BruceForstall** [comment]
Also fails on Windows

**Comment by BruceForstall** [comment]
Also fails in same way:
```
JIT\Regression\JitBlue\GitHub_20657\GitHub_20657\GitHub_20657.cmd
```

**Comment by BruceForstall** [comment]
For GitHub_20657, the issue is as follows.

First, note that the arm64 ABI states that registers v8-v15 are callee-saved, but _only the low 64 bits of the 128 bit register is callee-saved_. The high 64 bits are scratch. This is true for both Windows and Linux.

See https://docs.microsoft.com/en-us/cpp/build/arm64-windows-abi-conventions, http://infocenter.arm.com/help/topic/com.arm.doc.ihi0055b/IHI0055B_aapcs64.pdf.

In the test, function Oops(), with JitStressRegs=1, register v8/q8 is loaded with a value. In particular, the high 32-bits have the value "1.0". Then, we call the helper `CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE`. This does a lot of work, including invoking the JIT to compile and run the cctor. The JIT trashes q8/v8 (the high part anyway). After this call, the test checks the value to see it is what was expected, and it is not.

Without JitStressRegs=1, it appears the register is spilled around the helper call.

**Comment by CarolEidt** [comment]
On arm64/Windows, the hfa tests fail sporadically with `COMPlus_JitStress=2` and `COMPlus_JitStressRegs=1`. These tests pass reliably with #22257, but I suspect that is mostly coincidence.
I don't think these failures are directly related to the fact that these are HFA tests, since 1) We don't (yet) home any of these in vector registers, and 2) in the one case that fails the most often, all the HFAs are on the stack (not enregistered).
However, it appears that at least `System.SpanHelpers:IndexOf()` is keeping a vector live across a call. It's not (yet) clear why this is causing just these HFA tests to fail.

**Comment by CarolEidt** [comment]
As it happens, the support for partial spill is indeed enabled for Arm64 (see the definition of `FEATURE_PARTIAL_SIMD_CALLEE_SAVE` in target.h). In the `Oops` method if the GitHub_2067 test case, there's a previous call across which V08 is live, and its upper half is extracted from `v8` to `v9`.
However, the JIT only handles register candidate lclVar vectors that are live across a call (not tree temps), though AFAICT there is no fundamental requirement that dictates that large vector tree temps won't be. Across the call to `CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE`, the value of `V00` is loaded before the call, but it is not a register candidate, so it's a tree temp that doesn't get saved. I'm not certain why this hasn't happened on x64, but I suspect that it's rare to have a non-lclVar vector live across a call.

**Comment by BruceForstall** [comment]
Ok, the HFA failures look to be a different root cause. Let's use https://github.com/dotnet/coreclr/issues/22284 to track those failures.

**Comment by CarolEidt** [comment]
The GitHub_20657 failure ws fixed by #22311, and the hfa failures were fixed by #22313
